### ISSUE TITLE
[builder] add imgpkg as builder

### ISSUE THREAD

**Issue by cppforlife** [carvel-dev/kbld#56]
github.com/k14s/imgpkg is a builder that can create images from a set of files. let's have that as a backend. there probably needs to be two options: regular or bundle.

for bundles, i think we should have an option to run kbld to update images.yml

**Comment by voor** [comment]
This concept just blew my mind.

So instead of imgpkg push I would have a kbld configuration in my package cr that says "go to this folder for your bundle" and it would generate the bundle right there?

**Comment by cppforlife** [comment]
> So instead of imgpkg push I would have a kbld configuration in my package cr that says "go to this folder for your bundle" and it would generate the bundle right there?

right, for convenience however we would probably want to run kbld on the bundle itself (aka recursive call), but that requires a bit of gymnastics in non-trivial cases (e.g. kbld <(helm template ...)) and then we start getting into business of running other commands (which isnt a business i think we should be in).

**Comment by voor** [comment]
With helm we probably need better tutorials and documentation on how vendir can help and kapp-controller will do the helm template step for you.

**Comment by cppforlife** [comment]
> kapp-controller will do the helm template step for you.

yeah, but we still need kbld to collect images to write out .imgpkg/images.yml at the build time.

**Comment by voor** [comment]
> > kapp-controller will do the helm template step for you.
> 
> yeah, but we still need kbld to collect images to write out .imgpkg/images.yml at the build time.

:exploding_head: You're totally right, now I'm not sure how this will work
